console/program/src/data/plaintext/equal.rs

```diff
@@ -41,7 +41,18 @@ impl<N: Network> Equal<Self> for Plaintext<N> {
                 }
                 false => Boolean::new(false),
             },
-            (Self::Literal(..), _) | (Self::Struct(..), _) => Boolean::new(false),
+            (Self::Array(a, _), Self::Array(b, _)) => match a.len() == b.len() {
+                true => {
+                    // Recursively check each element for equality.
+                    let mut equal = Boolean::new(true);
+                    for (plaintext_a, plaintext_b) in a.iter().zip_eq(b.iter()) {
+                        equal &= plaintext_a.is_equal(plaintext_b);
+                    }
+                    equal
```

Comment:
```suggestion
                    if a.iter().zip_eq(b.iter()).all(|(plaintext_a, plaintext_b)| plaintext_a.is_equal(plaintext_b)) {
                        Boolean::new(true)
                    } else {
                        Boolean::new(false)
                    }
```
same principle as before